engine/src/execution_graph/logic_controllers/taskflow/kpair.h

```diff
@@ -1,10 +1,12 @@
 #pragma once
 
 #include "port.h"
+#include "kernel.h"
 
 namespace ral {
 namespace cache { 
 
+using kernel_pair = std::pair<kernel *, std::string>;
```

Comment:
why are we using a naked pointer here?

Reply:
excellent question. This line here was added so that it would compile. kernel_pair was originally only defined in another file, but for some reason the compiler was not seeing it, so i declared it here as well

Reply:
Let ebe more clear why is this a not a 
```
std::pair<std::shared_ptr<kernel>, std::string>
```

Reply:
I understood what you meant. Yes I agree that it should not be naked. I will add a task to fix this along with the implementation of the interfaces as mentioned in the previous comment

Reply:
I guess me saying `excellent question` was not a good answer. WHat i meant by that, is yes, i agree that it should not be a naked pointer. I dont know why it was originally done that way

Reply:
issue:  https://github.com/BlazingDB/blazingsql/issues/573